dali/operators/input/video_input.cc

```diff
@@ -0,0 +1,165 @@
+video file ends, the Operator will fetch the next one automatically from the top of the queue.
+Running the pipeline while there is no data for the ``fn.inputs.video`` to run results in an error.
+
+This operator takes only one video as and input (i.e. ``input_batch_size=1``) and will return
+batches of sequences. Every output batch will have the ``max_batch_size``, set during the Pipeline
+creation. When the number of frames in the video file does not allow to split the frames uniformly
+across batches, the last batch returned by this operator for a given video will be partial and the
+last sequence in this batch will be determined using ``last_sequence_policy`` parameter.
+For example::
+
+
+    This is a video that consists of 67 frames (every '-' is a frame):
+    -------------------------------------------------------------------
+
+
+    User decided that there shall be 5 frames per sequence and the last_sequence_policy='partial':
+    -------------------------------------------------------------------
+    [   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][   ][]
+    -------------------------------------------------------------------
+                      Since there are not enough frames, the last sequence comprises 2 frames.
+
+
+    The Pipeline has max_batch_size=3, therefore the operator will return 5 batches of sequences.
+    First 4 batches comprise 3 sequences and the last batch is partial and comprises 2 sequences.
+    ---------------   ---------------   ---------------   ---------------   -------
+    [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][]
+    ---------------   ---------------   ---------------   ---------------   -------
+
+
+    With the last_sequence_policy='pad', the last sequence of the last batch will be padded with 0:
+    ---------------   ---------------   ---------------   ---------------   -------000
+    [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][   ][   ]   [   ][   ]
+    ---------------   ---------------   ---------------   ---------------   -------000
+
+The difference between ``fn.inputs.video`` and ``fn.readers.video`` is that the former
+reads an encoded video from memory and the latter reads the encoded video from disk.
+
+The difference between ``fn.inputs.video`` and ``fn.decoders.video`` is that the former
+does not decode the whole video file in one go. This behaviour is needed for longer videos. E.g.
+5-min, 4k, 30fps decoded video takes about 1.7 TB of memory.
+
+This operator accepts most of the video containers and file formats. FFmpeg is used to parse
+the video container. In the situations, that the container does not contain required metadata
+(e.g. frames sizes, number of frames, etc...), the operator needs to find it out itself,
+which may result in a slowdown.
+)code")
+                .NumInput(0)
+                .NumOutput(1)
+                .AddArg("frames_per_sequence", R"code(
```

Comment:
True. On the other hand, I feel that `frames_per_sequence` is better, since `sequence_length` may be in seconds and we have a request for a variable frame rate videos to alternatively specify duration. I'll ask other guys in the team.

Reply:
My vote would go to sticking to `sequence_length` as well.

Reply:
Changed name to `sequence_length`